src/in-depth/machine-communication.md

```diff
@@ -230,10 +230,42 @@ This is how Visual Studio Code uses _ripgrep_ for its code search.
 
 ## How to deal with input piped into us
 
-<aside class="todo">
+Let's say we have a program that reads the number of words in a file:
 
-**TODO:**
-Talk about how work with stdin
-(see [#95](https://github.com/rust-cli/meta/issues/95))
+``` rust,ignore
+{{#include machine-communication-wc.rs}}
+```
 
-</aside>
+It takes the path to a file, reads it line by line, and counts the number of
+words separated by a space.
+
+When you run it, it outputs the total words in the file:
+
+``` console
+$ cargo run README.md
+Words in README.md: 47
+```
+
+But what if we wanted to count the number of words piped into the program?
+Rust programs can read data passed in via stdin with the
+[`std::io::stdin`](https://doc.rust-lang.org/std/io/fn.stdin.html) from the
```

Comment:
I feel like it might be a good idea to differentiate between [stdin (the function)](https://doc.rust-lang.org/std/io/fn.stdin.html) and [Stdin (the struct)](https://doc.rust-lang.org/std/io/struct.Stdin.html)
Maybe write something like:
```markdown
Rust programs can read data passed in via stdin with the with [the Stdin struct](https://doc.rust-lang.org/std/io/struct.Stdin.html) which you can obtain via [the stdin function](https://doc.rust-lang.org/std/io/fn.stdin.html) from the standard library. Similar to reading the lines of a file, it can read the lines from stdin.
```

Reply:
Thanks for this feedback, integrated it! I appreciate its clarity.

Reply:
No problem, glad to help!